### PR TITLE
Add build scripts, add docker image for building

### PR DESCRIPTION
**Pull request by andrebrait**
I've removed from the VCS every file that can be generated by building Adrenaline, and then I added all of them to the `.gitignore`. This way, simply building it won't make it seem like there's a number of changes to the code. The `.gitignore` also contains ignores for standard files generated from compiling C projects (`.o`, `.a`, etc.), ignores for Cmake-generated files, ignores for VS Code, etc.

I've also added shell scripts to make it easier to properly clean and rebuild the whole project. It also serves as documentation by explicitly telling you the steps required to build each part, and in which order that should happen.

`cef`'s Makefile will now also run the `release` goal of `systemctrl` so `psp-fixup-imports` will run when building the whole thing.

Additionally, it's nice to have an isolated build environment, or something that at least works as a documentation on how to achieve a working build environment, so I added a `Dockerfile` to do just that. It builds a Debian-based container with [pspdev (pre-built)](https://github.com/pspdev/pspdev), [psp-packer (built from source)](https://bitbucket.org/DaveeFTW/psp-packer), [Vita SDK (using VDPM)](https://github.com/vitasdk/vdpm), [vita-shader-collection (pre-built)](https://github.com/frangarcj/vita-shader-collection) and [libvita2d (fbo, built from source)](https://github.com/frangarcj/vita2dlib/tree/fbo).

The script `buildscripts/build.sh` builds the entire project, plus the `pkg` and `release` goals of some modules. The `buildscripts/build_docker.sh` script builds the Docker container image and then uses that to run `buildscripts/build.sh`, so the whole build runs inside the container. The image is cached by Docker, so it won't have to be rebuilt every time. In case rebuilding the image is desired (like if you want to update something), just pass `--no-cache` as argument to `build_docker.sh` and it will rebuild the whole thing, giving you a fully updated image.

I also added some explanation on this to the README.

I also moved some headers around so they would be in their shared places, added some functions to header files so the compiler wouldn't complain about implicit declarations. I removed the headers from the `vita-shader-collection` project because those will be expected to be installed to the Vita SDK folder.

There was some logic for calculating FPS that would be running whenever a shader was also running, and the actual displaying of the FPS was commented out, so I uncommented it and hid the whole thing (displaying the FPS + calculation) behind a preprocessing directive. A few CPU cycles saved that way, unless you define `SHOW_FPS` for the build :wink:

### CODE REVIEW

File: cef/systemctrl/main.c [hunk 1]
```diff
@@ -328,7 +328,7 @@ void OnSystemStatusIdle() {
 
 	// Set fake framebuffer so that cwcheat can be displayed
 	if (adrenaline->pops_mode) {
-		sceDisplaySetFrameBuf661((void *)0x0A000000, PSP_SCREEN_LINE, PSP_DISPLAY_PIXEL_FORMAT_8888, PSP_DISPLAY_SETBUF_NEXTFRAME);
+		sceDisplaySetFrameBuf661((void *)NATIVE_FRAMEBUFFER, PSP_SCREEN_LINE, PSP_DISPLAY_PIXEL_FORMAT_8888, PSP_DISPLAY_SETBUF_NEXTFRAME);
```
Comment (andrebrait) [inline cef/systemctrl/main.c:331]: Replaced magic number with constant

File: user/menu.c [hunk 2]
```diff
@@ -389,7 +389,7 @@ int AdrenalineDraw(SceSize args, void *argp) {
   vita2d_init();
   font = vita2d_load_default_pgf();
 
-  vita2d_texture *native_tex = vita2d_create_empty_texture_data(SCREEN_WIDTH, SCREEN_HEIGHT, CONVERT_ADDRESS(NATIVE_FRAMEBUFFER), SCE_GXM_TEXTURE_FORMAT_U5U6U5_BGR);
+  vita2d_texture *native_tex = vita2d_create_empty_texture_data(SCREEN_WIDTH, SCREEN_HEIGHT, (void*)CONVERT_ADDRESS(NATIVE_FRAMEBUFFER), SCE_GXM_TEXTURE_FORMAT_U5U6U5_BGR);
```
Comment (andrebrait) [inline user/menu.c:392]: Remove compiler warning about using `int` as a pointer without casting

**Comment by KuromeSan** [comment]
missing the 'btcnf', tool for building a bootconfig, which i actually cant find on the internet either.

**Comment by andrebrait** [comment]
@KuromeSan you mean in the resulting docker image?

**Pull request closed without merging**